### PR TITLE
ENH: estimate parameters of the Dirichlet distribution

### PR DESCRIPTION
**Pull request by argriffing**
Note that this is a multivariate distribution, so it is in a less-polished framework than the location-scale families of continuous distributions on the real line.

### CODE REVIEW

**Comment by ev-br** [comment]
This deserves way more attention than it received so far!

I tried to play with it, and I'm wondering about the return values: what's the rationale for returning loc and scale?

For univariate distributions, it seems to make sense: 
```
>>> from scipy.stats import gamma
>>> a = 0.5
>>> gdata = gamma.rvs(a, size=20, random_state=1234)
>>> res = gamma.fit(gdata)
>>> gamma.pdf(gdata, *res)
# works!
``` 
Here the signature is  `pdf(self, x, a, loc, scale)`. So far so good.

For `dirichlet`,  `pdf` does not accept loc and scale arguments (nor should it?), hence
```
>>> a = [0.5, 0.2]
>>> ddata = dirichlet.rvs(a, size=20, random_state=1234)
>>> res = dirichlet.fit(data)
>>> res
(array([ 4.70683424,  0.34123452]), array([ 0.,  0.]), array([ 1.,  1.]))
```

and

```
>>> dirichlet.pdf(data, res)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/br/installs/scipy-dev/scipy/build/testenv/lib/python2.7/site-packages/scipy/stats/_multivariate.py", line 846, in pdf
    alpha = _dirichlet_check_parameters(alpha)
  File "/home/br/installs/scipy-dev/scipy/build/testenv/lib/python2.7/site-packages/scipy/stats/_multivariate.py", line 656, in _dirichlet_check_parameters
    raise ValueError("All parameters must be greater than 0")
ValueError: All parameters must be greater than 0
```
or maybe

```
>>> dirichlet.pdf(data, *res)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
TypeError: pdf() takes exactly 3 arguments (5 given)
```

OK, there's an unrelated issue with "vectorization" with multiple values for `data`:

```
>>> dirichlet.pdf(data[0], res[0])
0.36047258089458101
```

but I guess it'd be nice if `dirichlet.pdf(data[0], res)` worked.

**Comment by argriffing** [comment]
It looks like the `fit` function was written to work with the numpy dirichlet rvs rather than with the scipy dirichlet pdf.  Unfortunately these interfaces are different.  Also a scipy dirichlet error message is buggy and also has misleading information...

**Comment by argriffing** [comment]
Closing this PR until the current behavior of `stats.dirichlet` has been bugfixed and more tests have been added.

**Comment by ev-br** [comment]
Could you summarize the problems / desiderata?

**Comment by argriffing** [comment]
> Could you summarize the problems / desiderata?

I'm making a PR...

**Comment by argriffing** [comment]
One of the problems is that I'd merged the enh:dirichlet PR before having unit-tested the error handling.  Sorry about that.

**Comment by argriffing** [comment]
> Could you summarize the problems / desiderata?

see https://github.com/scipy/scipy/pull/4984

**Comment by ev-br** [comment]
For the record, I think this should be resurrected once some variant of #4984 lands. 

My question about loc and scale stands irrespective of the transpose, and from a first glance I don't see a reason for returning `loc, scale = np.zeros(...), np.ones(...)` which are not going to be used anyway.
